app/models/usergroup.rb

```diff
@@ -36,25 +36,25 @@ class Usergroup < ActiveRecord::Base
   # This methods retrieves all user addresses in a usergroup
   # Returns: Array of strings representing the user's email addresses
   def recipients
-    all_users.map(&:mail).flatten.sort.uniq
+    all_users.map(&:mail).flatten.uniq.sort
```

Comment:
Take a look at the second answer there. The accepted answer uses an array that is randomized so that it has few or no duplicates, in that case there isn't much difference in run time. When there are duplicates, `uniq` reduces the number of the elements that need sorting.
`Array#uniq` checks for uniques using a hash table, so it doesn't care about the order the array is in. This is executed in O(n) time.
`Array#sort` uses quicksort, which usually runs in O(nlogn). This means that it is always slower then `uniq`. If `uniq` reduces the size of n, then sort will also benefit. If `sort` is run first, `uniq` has no way of knowing the array is sorted and will still create a hash table (rather then checking only every 2 consecutive items).